src/librustc_typeck/check/intrinsic.rs

```diff
@@ -78,6 +79,32 @@ pub fn check_intrinsic_type<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                                       it: &hir::ForeignItem) {
     let param = |n| tcx.mk_param(n, Symbol::intern(&format!("P{}", n)).as_interned_str());
     let name = it.name.as_str();
+    // The inner platform specific structure of a va_list.
+    let va_list_tag = tcx.lang_items().va_list().map(|did| {
```

Comment:
Is this correct? This is what I use to get the type of the inner `va_list` structure (`VaListImpl`) from the `VaList` lang item `DefId`. Is there a better way of doing this? Is it worthwhile?